java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -99,8 +105,31 @@ abstract class AbstractArcHost(
     // TODO: add lifecycle API for ArcHosts shutting down to cancel running coroutines
     private val scope = CoroutineScope(coroutineContext)
 
+    /** Supports asynchronous [ArcHostContext] serializations. */
+    private val channel: Channel<suspend () -> Unit> = Channel(Channel.UNLIMITED)
+    private val updateArcHostContextScope = CoroutineScope(updateArcHostContextCoroutineContext)
+
     init {
         initialParticles.toList().associateByTo(particleConstructors, { it.first }, { it.second })
+
+        /**
+         * Asynchronous [ArcHostContext] serializations in observed order in a hot [Flow].
+         * The [channel] will be cancelled when the [updateArcHostContextScope] is cancelled.
+         *
+         * TODO:
+         * make the channel per-Arc instead of per-Host for better serialization
+         * performance under multiple running and to-be-run Arcs.
+         */
+        channel.consumeAsFlow().onEach { it() }.launchIn(updateArcHostContextScope)
```

Comment:
Why do we need a flow here?

For a simple case like this, just using the channel directly seems fine:
```
val contextSerializationJob = scope.launch {
   channel.forEach { it() }
}
```

Reply:
What is `channel.forEach`?

The channel receiver side needs to do something like
```
scope.launch { while (true) channel.receive()() }
```
This is equivalent to using a hot flow to collect/receive element from the channel.
```
channel.consumeAsFlow().onEach { it() }.launchIn(scope)
``` 

I'm ok with the two styles and the latter looks a bit better to me.

Reply:
a ReceiveChannel is iterable, you should be able to do:
```
scope.launch {
  for(task in channel) {
    task()
  }
}
```

To me the intent here is much clearer than with the flow.

Reply:
hmm ok, i'll change it if it's clearer to read.

Reply:
I am willing to keep debating and be swayed in the other direction. 

I get the value of flows for the cold stream usage, but using them to wrap channels for simple scenarios like this feels less valuable to me.

For this *particular* use case, introducing flows feels like additional complexity and overhead for no perceived benefit... but it's definitely possible that I'm missing or forgetting something.

Reply:
For simple use case, channel or flow (wrap a channel) doesn't make big difference to me.
The flow just provides much flexibility like combining flows, flowOn to switch coroutine context, onCompletion to take care the tear-down jobs upon cancellation, etc.

For now, only send and receive-then-execute the task, so changing to use channel directly sounds reasonable to me.
In the future, if other modules need to listen to i.e. completion of serializations, a broadcasted channel as a flow is required for easier flow combination.

Reply:
Refactored to use Channel directly for the time being.